metrics_miscellany.org

```diff
@@ -1339,15 +1397,31 @@ __version__ = "0.3.1"
 ######################################################
 # Beginning of procedural version of gmm routines
 
-def gN(b):
-    """Averages of g_j(b).
+def gN(b,data):
+    """Averages of g_j(b,data).
 
     This is generic for data, to be passed to gj.
+
+    Parameters
+    ----------
+    b:numpy array, the initial value of b0
+    data: tuple, the data we'll input to estimate b0.
+
+    Returns
+    -------
+    Sample moments @ b
     """
-    e = gj(b)
+    e = gj(b,data)
 
     gN.N,gN.k = e.shape
-    gN.N = e.count()  # Allows for possibility of missing data
+
+    if hasattr(e,'count'):
```

Comment:
The function gj should be designed to return a pd.Series or pd.DataFrame, so if we don't have a count() method there's a deeper problem.  Maybe keep this assertion, and raise ValueError with a message to this effect?

Reply:
Got it, I'll make the correction in that part.